packages/@tailwindcss-postcss/src/index.ts

```diff
@@ -170,19 +170,26 @@ function tailwindcss(opts: PluginOptions = {}): AcceptedPlugin {
           // Register dependencies so changes in `base` cause a rebuild while
           // giving tools like Vite or Parcel a glob that can be used to limit
           // the files that cause a rebuild to only those that match it.
-          for (let { base, pattern } of context.scanner.globs) {
+          for (let { base: globBase, pattern } of context.scanner.globs) {
+            // Avoid adding a dependency on the base directory itself, since it
+            // causes Next.js to start an endless recursion if the `distDir` is
+            // configured to anything other than the default `.next` dir.
+            if (pattern === '*' && base === globBase) {
+              continue
+            }
```

Comment:
I feel like I could've made money on this being a problem _in something_ lol. I'm real happy the fix is this easy.

Reply:
Before this we explicitly added this kind of pattern for the CLI. 😅